Source/Core/Core/IOS/USB/Emulated/WiiSpeak.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Core/IOS/USB/Common.h"
+#include "Core/IOS/USB/Emulated/Microphone.h"
+#include "Core/System.h"
+
+namespace IOS::HLE::USB
+{
+struct WiiSpeakState
+{
+  bool sample_on;
+  bool mute;
+  int freq;
+  int gain;
+  bool ec_reset;
+  bool sp_on;
+};
+
+class WiiSpeak final : public Device
+{
+public:
+  WiiSpeak(EmulationKernel& ios);
+  ~WiiSpeak();
+
+  DeviceDescriptor GetDeviceDescriptor() const override;
+  std::vector<ConfigDescriptor> GetConfigurations() const override;
+  std::vector<InterfaceDescriptor> GetInterfaces(u8 config) const override;
+  std::vector<EndpointDescriptor> GetEndpoints(u8 config, u8 interface, u8 alt) const override;
+  bool Attach() override;
+  bool AttachAndChangeInterface(u8 interface) override;
+  int CancelTransfer(u8 endpoint) override;
+  int ChangeInterface(u8 interface) override;
+  int GetNumberOfAltSettings(u8 interface) override;
+  int SetAltSetting(u8 alt_setting) override;
+  int SubmitTransfer(std::unique_ptr<CtrlMessage> message) override;
+  int SubmitTransfer(std::unique_ptr<BulkMessage> message) override;
+  int SubmitTransfer(std::unique_ptr<IntrMessage> message) override;
+  int SubmitTransfer(std::unique_ptr<IsoMessage> message) override;
+
+private:
+  WiiSpeakState m_sampler{};
+
+  enum Registers
+  {
+    SAMPLER_STATE = 0,
+    SAMPLER_MUTE = 0x0c,
+
+    SAMPLER_FREQ = 2,
+    FREQ_8KHZ = 0,
+    FREQ_11KHZ = 1,
+    FREQ_RESERVED = 2,
+    FREQ_16KHZ = 3,  // default
+
+    SAMPLER_GAIN = 4,
+    GAIN_00dB = 0,
+    GAIN_15dB = 1,
+    GAIN_30dB = 2,
+    GAIN_36dB = 3,  // default
+
+    EC_STATE = 0x14,
+
+    SP_STATE = 0x38,
+    SP_ENABLE = 0x1010,
+    SP_SIN = 0x2001,
+    SP_SOUT = 0x2004,
+    SP_RIN = 0x200d
+  };
+
+  void GetRegister(const std::unique_ptr<CtrlMessage>& cmd) const;
+  void SetRegister(const std::unique_ptr<CtrlMessage>& cmd);
+  bool IsMicrophoneConnected() const;
+
+  EmulationKernel& m_ios;
+  const u16 m_vid = 0x57E;
+  const u16 m_pid = 0x0308;
+  u8 m_active_interface = 0;
+  bool m_device_attached = false;
+  bool init = false;
+  std::unique_ptr<Microphone> m_microphone{};
+  const DeviceDescriptor m_device_descriptor{0x12,  0x1,    0x200,  0,   0,   0,   0x10,
```

Comment:
```suggestion
  const DeviceDescriptor m_device_descriptor{
    .bLength = 0x12,
    .bDescriptorType = 0x01,
    .bcdUSB = 0x0200,
    .bDeviceClass = 0x00,
   ...
```

Reply:
AFAICT, this syntax isn't used in the current codebase for similar device classes. I wouldn't mind changing it in a follow up PR.